Automattic/newspack-plugin
# feat(ia): network settings

### All Submissions:






### Changes proposed in this Pull Request:



This PR moves "Settings" screens to tabs.  ( The previously merged "Foundational PR" is here: https://github.com/Automattic/newspack-plugin/pull/3421  ). Much of this new PR's code has been refactored based on newer techniques we're now using for these types of wizards.

The changes that need review are:

#### When the Site Role is "Node", then the "Node Settings" screen is moved to a tab.  Attached.

![epic-ia-network-settings-node-settings](https://github.com/user-attachments/assets/676f81b6-0eca-47f2-984b-95b3b943e83a)

#### When the Site Role is "Hub", then the "Distributor Settings" screen is moved to a tab.  Attached.

![epic-ia-network-settings-distributor-settings](https://github.com/user-attachments/assets/b149ba4b-0513-4dae-a2f5-a448c2a55370)

### How to test the changes in this Pull Request:

Note: if you see that any of the 3 CPTs (Nodes, Subscriptions, Orders) are not highlighting the correct submenu items, this is fixed in the Network plugin itself, so don't worry about that here.  ( See Network PR: https://github.com/Automattic/newspack-network/pull/146 ) 

1. Pull branch and make sure you have the Network plugin active.
2. Use wp-admin in your browser to click the different menu items in the Network menu verifying the admin header shows up and the correct submenu item is highlighted.
3. Do this twice: once as a Network "Node" and once as a Network "Hub".  
4. Verify the Settings pages (Distributor and Node Settings) are in tabs. 

### Other information:
